Contributing.md

```diff
@@ -1,30 +1,471 @@
-# Formatting
+# Contributing Guidelines
+
+You want to contribute to Tari, and we want you to contribute! So why is it sometimes hard to get this done?
+
+This is financial software, so we have to take precautions to ensure that harmful code doesn't slip in anywhere. 
+We know that _you_ are a great person and are just trying to help. But there are not-so-nice people 
+out there that want to see the world burn. These guidelines are primarily for those miscreants, but everyone 
+needs to 
+follow them.
+
+These guidelines are not cast in stone, or should we say, not committed to the blockchain. We are constantly 
+evaluating what works, what doesn't work and will update the guidelines from time to time as we learn. We also draw 
+inspiration from other successful open-source projects and adapt the best ideas for our purposes.
+
+The goals of having these guidelines are fourfold:
+1. To maintain a **secure and reliable codebase**. This is paramount.
+2. To maintain **high-quality code**. This means thinking hard about 
+  * fostering a fantastic developer experience, 
+  * by writing beautiful, ergonomic APIs, 
+  * using clean patterns, 
+  * writing excellent documentation, 
+  * producing very high code coverage on tests, 
+  * having tests run in a reasonable time, 
+  * and keeping technical debt to a minimum.
+3. To keep the code **open** (as in free).
+4. To keep the code open (as in **encouraging**). If we achieve #2, this comes for free. But have fun doing it. 
+
+These guidelines are split up into a few main topics:
+
+* [The Release Process](#the-tari-release-process). This explains how the release schedule for Tari works post-mainnet 
+  launch. 
+* The [Feature RFC process](#feature-rfc-process) describes how new features get into Tari, from conception 
+  to implementation.
+* [Pull requests](#pull-requests). This section offers guidelines on how to get your code merged into the code base 
+  with the minimum of fuss. 
+* [Code reviews](#code-reviews) are integral to keeping the code secure and performant. This section 
+  offers tips for effective preventing bugs from scuttling past your gimlet gaze. 
+* Bug bounty and disclosure policy.
+
+# The Tari release process
+
+The Tari release process draws inspiration from the Rust [compiler release process] and borrows some of its key 
+ideas.
+
+The process is well-suited for a large open-source project that desires a regular release schedule but also needs 
+the flexibility of having features being "in development" over several cycles.
+
+## Release branches
+
+Post mainnet, there are 4 release branches in the Tari repo:
+
+* `mainnet` - the code on this branch represents the latest "official release". Almost every commit on this branch 
+  will be tagged with a version number.
+* `development` - this is the cuttingest-of-edge branch. Almost all new code comes in via the `development` branch.
```

Comment:
It's not clear which testnet this relates to in the next section. From the ordering it looks like `stagenet`, but it's probably likely to be `dibbler`

Reply:
It is also not clear/discussed how code in `development` is migrated onto other branches.
- `dibbler` [with feature gates] (cherry picked only?) moves to `nextnet` moves to `mainnet`
- `development` [with feature gates] (cherry picked only?) moves to `dibbler` only or to `nextnet` as well?